src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -324,6 +321,13 @@ private async Task<WorkUnitResult> ExecuteTask(TaskExecutionMode mode, Lookup lo
 
                 WorkUnitResult aggregateResult = new WorkUnitResult();
 
+                // Some tests do not provide an actual taskNode; checking if _taskNode == null prevents those tests from failing.
+                long startTime = -1;
+                if (MSBuildEventSource.Log.IsEnabled())
+                {
+                    startTime = MSBuildEventSource.GetThreadTime();
+                    MSBuildEventSource.Log.ExecuteTaskStart(_taskNode?.Name);
```

Comment:
I think logging the task ID would be better. It would help users correlate the msbuild log task started / ended events and the ETW events.

Reply:
I don't see the use case for this? The original issue mentioned wanting to see `Done executing task "CL" in 15 seconds` or the equivalent to help with investigating slow builds. That means you would need to know what function (its name) was slow but no more.

Also, I didn't see an ID attribute of _taskNode.

Reply:
The issue refers to tasks that run multiple times in a build (e.g. for each project), and the user needs to differentiate them (otherwise, for singular tasks, they can just in the existing perf summary). I think it'd be helpful for them to be able to place the task in the event hierarchy, using the task id, so they can know which project is running that task. TaskBuilder should have the id somewhere, it's the one that logs task started / ended events, which contain the ID.